### ISSUE TITLE
Mismatched `types.Header` after v1.11.0 release for evm forks using `go-ethereum`

### ISSUE THREAD

**Issue by kaladinlight** [trezor/blockbook#914]
`WithdrawalsHash` was added to the block header in `v1.11.0` https://github.com/ethereum/go-ethereum/compare/v1.10.26...v1.11.0#diff-e177ddb5610d1f3426c15cb512999613785b20838954e48187883e6263bb282bR91

BSC, Polygon, and Optimism were set up to use the `go-ethereum` module provided by Ethereum which worked fine before `v1.11.0`. With the `v1.11.0` release, some of these evm forks no longer match up producing incorrect block hashes via the `Header.Hash()` function.

BSC (Needs Update): https://github.com/bnb-chain/bsc/blob/master/core/types/block.go#L100
Polygon (Not Merged, Needs Update): https://github.com/0xPolygon/go-ethereum/blob/master/core/types/block.go#L70
Optimism Bedrock (No Update Needed): https://github.com/ethereum-optimism/op-geth/blob/optimism/core/types/block.go#L65

**Comment by kaladinlight** [comment]
I am not having luck with aliased replace directive due to types not matching with different module names:

```
replace (
	github.com/bnb-chain/bsc => github.com/ethereum/go-ethereum v1.10.26
)
```

![image](https://user-images.githubusercontent.com/35275952/231890202-d61bf00c-d1ba-4e6e-aa55-f0d0fccfda31.png)

**Comment by kaladinlight** [comment]
Other thoughts:
- vendoring copies of the evm forks?
- maintaining a fork of the fork with updated module name
- go workspaces?

I haven't really dealt with this problem with go modules so I'm not sure if I am overlooking something obvious, but multiple versions of the same packages seem like an inherent complication...

**Comment by kaladinlight** [comment]
Disregard... The new field is marked optional for the rlp encoding, non breaking.

**Comment by martinboehm** [comment]
@kaladinlight  Hi. I am kind of lost. What is the issue and why we can disregard it?

**Comment by kaladinlight** [comment]
> @kaladinlight Hi. I am kind of lost. What is the issue and why we can disregard it?

Yeah sorry, so the issue was that I was running an older version of the ethereum blockbook that was still on `go-ethereum v1.10.26`. After the shanghai upgrade there was additional metadata added to the block headers `WithdrawalsHash` that caused incorrect block hashes when calling `header.Hash()`. I upgraded my eth blockbook so it was using `go-ethereum v1.11.4` which resolved the issue for eth.

I was then thinking that the additional header field added would mess up the hashes for bsc since it is just using `go-ethereum` as well, but turns out that was not the case as they added it using `rlp:optional` tag:
![image](https://user-images.githubusercontent.com/35275952/232102206-96fd16c8-2fe5-461e-b578-50e88bbc0525.png)

So basically it was just me freaking out over nothing, but I thought we would need to find a way to add another version of `go-ethereum` frozen at `1.10.26` for the new evm chains being added.

Hopefully that helps clarify, let me know if you have more questions.
